apps/music/js/metadata/flac.js

```diff
@@ -36,37 +37,64 @@ var FLACMetadata = (function() {
     // First four bytes are "fLaC" or we wouldn't be here.
     blobview.seek(4);
 
-    metadata.tag_format = 'flac';
-    return findVorbisCommentBlock(blobview).then(function(block) {
-      if (!block) {
-        return metadata;
+    return new Promise(function(resolve, reject) {
+      metadata.tag_format = 'flac';
+      var has_vorbis_comment = false;
+      var has_picture = false;
+
+      function processBlock(block) {
+        if (!block) {
+          resolve(metadata);
+          return;
+        }
+
+        if (block.block_type === 4) {
+          readAllComments(block.view, metadata);
+          has_vorbis_comment = true;
+
+        } else if (block.block_type === 6) {
+          LazyLoader.load('js/metadata/vorbis_picture.js', function() {
+            metadata.picture = VorbisPictureComment.readPicFrame(block.view);
+            metadata.picture.start += block.view.viewOffset;
+            metadata.picture.end += block.view.viewOffset;
+          });
+          has_picture = true;
+
+        }
+
+        if (!has_vorbis_comment || !has_picture) {
+          try {
+            block.view.advance(block.length - block.view.index);
+            findMetadataBlocks(block.view, processBlock);
```

Comment:
Instead of seeking and finding another metadata block here, let's do this in `findMetadataBlocks`.